### PR TITLE
Release process definition and implementation

### PR DESCRIPTION
**Pull request by roytman**
## Why are these changes needed?

We need to define and after that automate the project release process.

## Related issue number (if any).
https://github.com/IBM/data-prep-kit/issues/294

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+The process of creating a release is described in this document.
+Note: below we list the manual steps to create a new release, later some of them will be automated by GitHub actions.
+
+Replace `X.Y.Z` with the version to be released.
+
+## 1. Update the .make.versions file
+
+**Question:** should we use the same versions for release, and for shared python modules and images, such as 
+`DPK_LIB_VERSION`, `DPK_LIB_KFP_VERSION`, `DPK_LIB_KFP_VERSION_v2`, `DPK_LIB_KFP_SHARED`, `KFP_DOCKER_VERSION`, `KFP_DOCKER_VERSION_v2`?
+If YES, maybe we can combine all these variables into a single one.
```
Comment (revit13) [inline RELEASE.md:12]: Does it makes sense to have `0.0.0` tag to all in dev branch and update it to the release tag upon creating a new release in the release branch (assuming their tag version is as the relase version tag)? there is no meaning to the tag version in dev as we build everything locally.

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+- In the [.make.versions](.make.versions) file update versions of the shared components `DPK_LIB_VERSION`, 
+`DPK_LIB_KFP_VERSION`, `DPK_LIB_KFP_VERSION_v2`, `DPK_LIB_KFP_SHARED`, `KFP_DOCKER_VERSION`, `KFP_DOCKER_VERSION_v2` to the desired ones  
+- Update transformers versions
+- Set `RELEASE_VERSION_SUFFIX` to the empty string, or comment out the entry.
+
+## 2. Update versions
+From the project root directory execute
+```shell
+make versions
```
Comment (revit13) [inline RELEASE.md:22]: I think its `make set-versions`

File: RELEASE.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+## 1. Update the .make.versions file
+
+**Question:** should we use the same versions for release, and for shared python modules and images, such as 
+`DPK_LIB_VERSION`, `DPK_LIB_KFP_VERSION`, `DPK_LIB_KFP_VERSION_v2`, `DPK_LIB_KFP_SHARED`, `KFP_DOCKER_VERSION`, `KFP_DOCKER_VERSION_v2`?
+If YES, maybe we can combine all these variables into a single one.
+
+- In the [.make.versions](.make.versions) file update versions of the shared components `DPK_LIB_VERSION`, 
+`DPK_LIB_KFP_VERSION`, `DPK_LIB_KFP_VERSION_v2`, `DPK_LIB_KFP_SHARED`, `KFP_DOCKER_VERSION`, `KFP_DOCKER_VERSION_v2` to the desired ones  
+- Update transformers versions
```
Comment (revit13) [inline RELEASE.md:16]: Its not clear how the transforms release process is done. should it be independent of the repo release? if so, we should not update the versions in this process.

File: RELEASE.md (hunk 1 above)
Comment (revit13) [reply RELEASE.md:12]: BTW I think that the process of updating the versions in .make.versions should be done in the release branch so the .make.versions in dev branch should remain `0.0.0`...

Comment (roytman) [reply RELEASE.md:12]: > there is no meaning to the tag version in dev as we build everything locally.

it is not exactly true, for pipeline tests on a real cluster we need relevant images.

**Pull request closed without merging**